buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/transform/RestTestTransform.java

```diff
@@ -17,9 +17,16 @@
 
     /**
      * Transform the Json structure per the given {@link RestTestTransform}
-     * Implementations are expected to mutate the node (and/or it's parent) to satisfy the transformation.
+     * Implementations are expected to mutate the parent to satisfy the transformation.
      *
-     * @param node The node to transform. This may also be the logical parent of the node that should be transformed.
+     * @param parent The parent of the node to transform.
      */
-    void transformTest(T node);
+    void transformTest(T parent);
+
+    /**
+     * @return The test name to to apply this transformation to. if {@code null} will apply to any test irregardless of the name.
+     */
+    default String getTestName() {
```

Comment:
This seems like it could be better expressed as something like `boolean shouldApply(TestContext test)`. Where `TestContext` would include the test name, and potentially extended later to include parent, etc. Transformer would then simply call this to see if it should run and each implementation could use whatever implementation they want to determine if the transform is applicable or not. Default impl could just return `true`.

Reply:
++ I like that.  There is possibility that sometime in the future I will need to apply/notApply based on more conditions besides the test name.

Reply:
done here: [`ef6c666` (#68338)](https://github.com/elastic/elasticsearch/pull/68338/commits/ef6c66604b92a1e8e970dce57e4a1b020be71754) (header fixed in different commit)